src/main/java/org/openrewrite/java/migrate/UpgradeJavaVersion.java

```diff
@@ -65,24 +64,17 @@ public TreeVisitor<?, ExecutionContext> getVisitor() {
                 }
                 SourceFile source = (SourceFile) tree;
 
-                Optional<JavaVersion> maybeJavaVersion = source.getMarkers().findFirst(JavaVersion.class);
-                if (maybeJavaVersion.isPresent() && maybeJavaVersion.get().getMajorVersion() >= version) {
-                    // No change if try to downgrade java version, or on same java version.
-                    return source;
-                }
-
                 if (source instanceof G.CompilationUnit && new IsBuildGradle<ExecutionContext>().visit(source, ctx) != source) {
                     source = (SourceFile) new UpdateJavaCompatibility(version, null, null, false).getVisitor().visitNonNull(source, ctx);
                 } else if (source instanceof Xml.Document) {
                     source = (SourceFile) new MavenUpdateJavaVersionVisitor().visitNonNull(source, ctx);
                 }
 
-                if (maybeJavaVersion.isPresent()) {
-                    source =
-                        source.withMarkers(source.getMarkers().setByType(updatedMarkers.computeIfAbsent(maybeJavaVersion.get(),
-                        m -> m.withSourceCompatibility(newVersion).withTargetCompatibility(newVersion))));
+                Optional<JavaVersion> maybeJavaVersion = source.getMarkers().findFirst(JavaVersion.class);
```

Comment:
As you recall, `JavaVersion` wouldn't ever be present on a build file, so there still wouldn't need to be a reason to update it.

Reply:
But this visitor isn't limited to only build files right? I thought we're updating these markers on Java classes too here, but maybe I'm switching between too many tasks.

Reply:
Ahh, no you're right. I'm not awake enough. 😅